src/cpu/map.cc

```diff
@@ -0,0 +1,135 @@
+#include "map.hh"
+
+#include <cstdlib>
+#include <ctime>
+#include <iostream>
+#include <ncurses.h>
+#include <thread>
+
+Map::Map(size_t height, size_t width)
+    : height_{height}
+    , width_{width}
+    , map_(height * width)
+{
+    initscr();
+
+    std::srand(std::time(nullptr));
+    for (size_t i = 0; i < height_ * width_; i++)
+    {
+        map_[i] = Cell(std::rand() / ((RAND_MAX + 1u) / 2));
+    }
+}
+
+Map::~Map()
+{
+    endwin();
+}
+
+int Map::number_of_alive_neighbours(size_t j, size_t i) const
+{
+    int nb = 0;
+
+    for (size_t y = j - 1; y < j + 2; y++)
+    {
+        for (size_t x = i - 1; x < i + 2; x++)
+        {
+            if (is_valid_coord(y, x) && y != j && x != i)
```

Comment:
> The simplest way how to make a world infinite is make it cyclic.

Il faudrait faire un modulo. Toutes les cellules ont 8 voisins, même la cellule en (0, 0)

Reply:
fixed in #3